mne/beamformer/_rap_music.py

```diff
@@ -0,0 +1,238 @@
+"""Compute a Recursively Applied and Projected MUltiple
+Signal Classification (RAP-MUSIC).
+"""
+
+# Authors: Yousra Bekhti <yousra.bekhti@gmail.com>
+#          Alexandre Gramfort <alexandre.gramfort@telecom-paristech.fr>
+#
+# License: BSD (3-clause)
+
+import numpy as np
+from scipy import linalg
+
+from ..io.pick import pick_channels_evoked
+from ..minimum_norm.inverse import _check_reference
+from ..cov import compute_whitener
+from ..utils import logger, verbose
+from ..dipole import Dipole
+from ._lcmv import _prepare_beamformer_input, _setup_picks
+
+
+@verbose
+def _apply_rap_music(data, info, tmin, forward, noise_cov,
+                     signal_ndim=15, n_dipoles=5, picks=None,
+                     return_explained_data=False, verbose=None):
+    """RAP-MUSIC for evoked data
+
+    Parameters
+    ----------
+    data : array or list / iterable
+        Evoked data.
+    info : dict
+        Measurement info.
+    tmin : float
+        Time of first sample.
+    forward : dict
+        Forward operator.
+    noise_cov : Covariance
+        The noise covariance.
+    signal_ndim : int
+        The dimension of the subspace spanning the signal.
+        The default value is 15.
+    n_dipoles : int
+        The number of dipoles to estimate.
+    picks : array-like of int | None
+        Indices (in info) of data channels. If None, MEG and EEG data channels
+        (without bad channels) will be used.
+    return_explained_data : bool
+        If True, the explained data is returned as an array.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see mne.verbose).
+
+    Returns
+    -------
+    dipoles : instance of Dipole
```

Comment:
Ok, I was following the current Dipole class that for me was not taking only one dipole:

      """Dipole class

      Used to store positions, orientations, amplitudes, times, goodness of fit
      of dipoles, typically obtained with Neuromag/xfit, mne_dipole_fit
      or certain inverse solvers.
    
      Parameters
      ----------
      times : array, shape (n_dipoles,)
            The time instants at which each dipole was fitted.
      pos : array, shape (n_dipoles, 3)
            The dipoles positions in meters
      ...

I'll change for a list of instances of Dipole then.